File: video-series-mono/src/main/java/org/jesperancinha/video/event/VideoSeriesQueryHandler.java
package org.jesperancinha.video.event;

import org.axonframework.eventhandling.EventHandler;
import org.jesperancinha.video.domain.VideoSeries;
import org.springframework.stereotype.Component;

import org.jesperancinha.video.repository.VsaRepository;

@Component
public class VideoSeriesQueryHandler {

    private final VsaRepository vsaRepository;

    public VideoSeriesQueryHandler(VsaRepository vsaRepository) {
        this.vsaRepository = vsaRepository;
    }

    @EventHandler
    public void on(VideoSeriesEvent event) {
        vsaRepository.saveVideoSeriesHistory(new VideoSeries(
                event.getId(),
                event.getName(),
                event.getVolumes(),
                event.getCashValue(),
                event.getGenre())
        );
    }
}

File: video-series-mono/src/main/kotlin/org/jesperancinha/video/event/VideoSeriesQueryHandler.kt
package org.jesperancinha.video.event

import org.axonframework.eventhandling.EventHandler
import org.jesperancinha.video.domain.VideoSeries
import org.jesperancinha.video.repository.VsaRepository
import org.springframework.stereotype.Component

@Component
class VideoSeriesQueryHandler(private val vsaRepository: VsaRepository) {
    @EventHandler
    fun on(event: VideoSeriesEvent) {
        vsaRepository.saveVideoSeriesHistory(
            VideoSeries(
                event.id,
                event.name,
                event.volumes,
                event.cashValue,
                event.genre
            )
        )
    }
}